native/src/components/ExportEventButton.tsx

```diff
@@ -0,0 +1,117 @@
+  const { t } = useTranslation('events')
+  const theme = useTheme()
+  const showSnackbar = useSnackbar()
+
+  const [eventExported, setEventExported] = useState<boolean>(false)
+  const [showCalendarChoiceOverlay, setShowCalendarChoiceOverlay] = useState<boolean>(false)
+  const [calendars, setCalendars] = useState<Calendar[]>()
+
+  const saveEventToCalendar = async (calendarId: string | undefined): Promise<void> => {
+    await RNCalendarEvents.saveEvent(event.title, {
+      startDate: event.date.startDate.toISOString(),
+      endDate: event.date.endDate.toISOString(),
+      allDay: event.date.allDay,
+      location: event.location?.fullAddress,
+      description: event.excerpt,
+      calendarId,
+    })
+      .then(id => {
+        setEventExported(true)
+        showSnackbar({
+          text: t('added'),
+          positiveAction: {
+            label: t('goToCalendar'),
+            onPress: () => {
+              if (Platform.OS === 'ios') {
+                const appleRefDate = moment('Jan 1 2001', 'MMM DD YYYY')
+                const secondsSinceRefDate = event.date.startDate.diff(appleRefDate, 'seconds')
+                Linking.openURL(`calshow:${secondsSinceRefDate}`)
+              } else if (Platform.OS === 'android') {
+                Linking.openURL(`content://com.android.calendar/events/${id}`)
+              }
+            },
+          },
+        })
+      })
+      .catch(() => {
+        showSnackbar({ text: 'generalError' })
+      })
+  }
+
+  const checkCalendarsAndSaveEvent = async (): Promise<void> => {
+    let permission = await RNCalendarEvents.checkPermissions()
+    if (permission === 'undetermined') {
+      permission = await RNCalendarEvents.requestPermissions()
+    }
+    if (permission === 'denied' || permission === 'restricted') {
+      showSnackbar({
+        text: 'noCalendarPermission',
+        positiveAction: {
+          label: t('settings'),
+          onPress: Linking.openSettings,
+        },
+      })
+      return
+    }
+    const editableCalendars = (await RNCalendarEvents.findCalendars()).filter(cal => cal.allowsModifications)
+    if (editableCalendars.length === 0) {
+      showSnackbar({ text: 'noCalendarFound' })
+    } else if (editableCalendars.length === 1) {
+      saveEventToCalendar(editableCalendars[0]?.id)
+    } else {
+      setCalendars(editableCalendars)
+      setShowCalendarChoiceOverlay(true)
+    }
+  }
+
+  return (
+    <>
+      {!!calendars?.length && (
+        <CalendarChoice
+          closeOverlay={() => setShowCalendarChoiceOverlay(false)}
+          overlayVisible={showCalendarChoiceOverlay}
+          chooseCalendar={async (id: string) => {
+            setShowCalendarChoiceOverlay(false)
+            await saveEventToCalendar(id).catch(() => {
+              showSnackbar({ text: 'generalError' })
+            })
+          }}
+          calendars={calendars}
+        />
+      )}
+      <Button
```

Comment:
Optional since this was used in a few places before already: We should probably create a custom `TextButton` component at some time since we use the same code/styling in quite a few places. Margin is the only property specific to the use here which could be applied with styled-components as well.

Reply:
Great idea, I'll make a ticket for that!